models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055__denominator.sql

```diff
@@ -0,0 +1,208 @@
+
+    select patient_id
+         , coalesce(encounter.encounter_start_date,encounter.encounter_end_date) as min_date
+         , coalesce(encounter.encounter_end_date,encounter.encounter_start_date) as max_date
+    from {{ref('quality_measures__stg_core__encounter')}} encounter
+    inner join {{ref('quality_measures__int_cbe0055__performance_period')}} as pp
+        on coalesce(encounter.encounter_end_date,encounter.encounter_start_date) >= pp.performance_period_begin
+        and  coalesce(encounter.encounter_start_date,encounter.encounter_end_date) <= pp.performance_period_end
+    where lower(encounter_type) in (
+          'home health'
+        , 'office visit'
+        , 'outpatient'
+        , 'outpatient rehabilitation'
+        , 'telehealth'
+     )
+
+)
+
+, procedure_encounters as (
+
+    select 
+          patient_id
+        , procedure_date as min_date
+        , procedure_date as max_date
+    from {{ref('quality_measures__stg_core__procedure')}} proc
+    inner join {{ref('quality_measures__int_cbe0055__performance_period')}}  as pp
+        on procedure_date between pp.performance_period_begin and  pp.performance_period_end
+    inner join visit_codes
+        on coalesce(proc.normalized_code,proc.source_code) = visit_codes.code
+
+)
+
+, claims_encounters as (
+    
+    select patient_id
+    , coalesce(claim_start_date,claim_end_date) as min_date
+    , coalesce(claim_end_date,claim_start_date) as max_date
+    from {{ref('quality_measures__stg_medical_claim')}} medical_claim
+    inner join {{ref('quality_measures__int_cbe0055__performance_period')}}  as pp on
+        coalesce(claim_end_date,claim_start_date)  >=  pp.performance_period_begin
+         and coalesce(claim_start_date,claim_end_date) <=  pp.performance_period_end
+    inner join  visit_codes
+        on medical_claim.hcpcs_code= visit_codes.code
+
+)
+
+, all_encounters as (
+
+    select *, 'v' as visit_enc,cast(null as {{ dbt.type_string() }}) as proc_enc, cast(null as {{ dbt.type_string() }}) as claim_enc
+    from visits_encounters
+
+    union all
+
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc, 'p' as proc_enc, cast(null as {{ dbt.type_string() }}) as claim_enc
+    from procedure_encounters
+
+    union all
+    
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc,cast(null as {{ dbt.type_string() }}) as proc_enc, 'c' as claim_enc
+    from claims_encounters
+
+)
+
+, encounters_by_patient as (
+
+    select patient_id,min(min_date) min_date, max(max_date) max_date,
+        concat(concat(
+            coalesce(min(visit_enc),'')
+            ,coalesce(min(proc_enc),''))
+            ,coalesce(min(claim_enc),'')
+            ) as qualifying_types
+    from all_encounters
+    group by patient_id
+
+)
+
+, diabetics_codes as (
+
+    select
+          code
+        , code_system
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in (
+          'diabetes visit'
```

Comment:
This concept should be included with the other visit codes CTE since these are procedure codes. You can confirm that the codes in this concept are part of the "Patient encounter during the performance period" logic for the denominator in the measure definition.